### PR TITLE
fix(standalone): write `risedev-env` file in standalone mode

### LINKED ISSUE #17090: cannot `./risedev slt` after `./risedev p` since we forward impl to standalone
This is because `./risedev p` does not dump the env vars that will be used by `./risedev slt`. This was previously done in https://github.com/risingwavelabs/risingwave/pull/11660 but regressed after #15897.

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Resolve #17090.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by kwannoel — APPROVED** [review]
LGTM, thanks

**Comment by kwannoel** [comment]
We can test if it works by replacing `single-node`'s adhoc writing of the `risedev-env` file, inside `run-e2e-test.sh`.

**Comment by BugenZhao** [comment]
> We can test if it works by replacing `single-node`'s adhoc writing of the `risedev-env` file, inside `run-e2e-test.sh`.

Updated. Thanks for the reminder.

**Review by xxchan — APPROVED** [review]
❤️

File: ci/scripts/single-node-utils.sh [hunk 1]
```diff
@@ -5,6 +5,7 @@ set -euo pipefail
 export RW_PREFIX=$PWD/.risingwave
 export PREFIX_BIN=./target/debug
 export PREFIX_LOG=$RW_PREFIX/log
+export RISEDEV=1 # as if we are running in RiseDev
```
Comment (xxchan) [inline ci/scripts/single-node-utils.sh:8]: PREFIX_CONFIG is not set

**Comment by xxchan** [comment]
🤪
image

**Pull request merged**